### PR TITLE
2294 by Inlead: Various optimisation for ding_mobilesearch.

### PR DESCRIPTION
**Pull request by inlead**
http://platform.dandigbib.org/issues/2294

### CODE REVIEW

File: modules/ding_mobilesearch/ding_mobilesearch.install [hunk 1]
```diff
@@ -47,3 +47,30 @@ function ding_mobilesearch_uninstall() {
...
+  // Fetching flag options.
+  $options = db_select('flags', 'f')
+    ->fields('f', array('options'))
+    ->condition('name', 'push_to_mongo')
+    ->execute()
+    ->fetchField();
+
+  $options = unserialize($options);
+  // Setting local admin role.
+  $options['roles']['flag'][] = '6';
```
Comment (cableman) [inline modules/ding_mobilesearch/ding_mobilesearch.install]: Are we sure that the id "6" will match the right roll across all DDB sites?

File: modules/ding_mobilesearch/ding_mobilesearch.module [hunk 2]
```diff
@@ -267,13 +267,26 @@ function ding_mobilesearch_ding_mobilesearch_node_export_mapping($node) {
...
             'name' => $field['label'],
             'value' => array(),
             'attr' => array(),
           );
+
           foreach ($items as $item) {
-            $mapping['fields'][$machine_name]['value'][] = $item['target_id'];
+            $entity = entity_load_single($target, $item['target_id']);
+            if ($entity) {
+              // TODO: Consider taxonomies as well.
```
Comment (cableman) [inline modules/ding_mobilesearch/ding_mobilesearch.module]: Please remove todo and create issue if needed or fix the todo.

File: modules/ding_mobilesearch/ding_mobilesearch.install (hunk 1 above)
Comment (guddo) [inline modules/ding_mobilesearch/ding_mobilesearch.install]: Yes, this will always match the "local administrator" role.

**Comment by cableman** [comment]
Reviewed and awaiting release

**Pull request merged**